### ISSUE TITLE
jQuery.Deferred exception: self.dropdowns is undefined toggleDropDown in javascript console

### ISSUE THREAD

**Issue by rittmeier** [ether/etherpad-lite#4132]
**Describe the bug**
I run version 1.8.4 (75ecb3f) in a dev instance. Opening a pad gives a message in the javascript console. There are no other side effects yet. The message is:
>jQuery.Deferred exception: self.dropdowns is undefined toggleDropDown@http://sub.domain.tld:9001/javascripts/lib/ep_etherpad-lite/static/js/pad_utils.js?callback=require.define&v=2bef60d6:1802:21
showModal@http://sub.domain.tld:9001/javascripts/lib/ep_etherpad-lite/static/js/pad_utils.js?callback=require.define&v=2bef60d6:2137:18
connected@http://sub.domain.tld:9001/javascripts/lib/ep_etherpad-lite/static/js/pad.js?callback=require.define&v=2bef60d6:5146:17
handleChannelStateChange@http://sub.domain.tld:9001/javascripts/lib/ep_etherpad-lite/static/js/pad.js?callback=require.define&v=2bef60d6:762:27
setChannelState@http://sub.domain.tld:9001/javascripts/lib/ep_etherpad-lite/static/js/pad.js?callback=require.define&v=2bef60d6:3757:17
setUpSocket@http://sub.domain.tld:9001/javascripts/lib/ep_etherpad-lite/static/js/pad.js?callback=require.define&v=2bef60d6:3396:20
mightThrow@http://sub.domain.tld:9001/javascripts/lib/ep_etherpad-lite/static/js/ace2_common.js?callback=require.define&v=2bef60d6:3910:29
resolve/</process<@http://sub.domain.tld:9001/javascripts/lib/ep_etherpad-lite/static/js/ace2_common.js?callback=require.define&v=2bef60d6:3978:12
 undefined ace2_common.js:4186:18

**To Reproduce**
1. Checkout current develop branch
2. run bin/run.sh
3. Open etherpad in browser
4. Create pad
5. open javascript console

**Expected behavior**
no such message in the js consle

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser FF
 - Version 77.0.1

**Comment by JohnMcLear** [comment]
I'm 99% sure this isn't a problem with etherpad as tests are all passing (less docker).  

Is there something specific to your environment?

**Comment by rittmeier** [comment]
The box runs on Debian 10.4 (Buster). I just tried again, so cloned the current develop (168737b). Change the port in settings.json and then started using `./bin/run.sh -s settings.json`. Same entry in javascript console.

**Comment by rittmeier** [comment]
[console-edge.log](https://github.com/ether/etherpad-lite/files/4840627/console-edge.log)
Similiar message when accessing from Microsoft Edge 83.

**Comment by webzwo0i** [comment]
postAceInit that calls padeditbar.init seems to be called to late

**Comment by webzwo0i** [comment]
but it should be called soon enough... https://github.com/ether/etherpad-lite/blob/develop/src/static/js/pad.js#L530

**Comment by webzwo0i** [comment]
padeditbar.init(); that initializes the dropdowns is called from https://github.com/ether/etherpad-lite/blob/develop/src/static/js/ace.js#L218 but probably on event loop's next tick

**Comment by webzwo0i** [comment]
since latest jquery-update, but it's not covered by our frontend tests so might be broken before

**Comment by JohnMcLear** [comment]
Steps to replicate?  I can't replicate in develop...

**Comment by webzwo0i** [comment]
weird. I see it in firefox 68.9.0esr-1~deb10u1 or chromium 80.0.3987.162-1~deb10u1 while running the tests.
it's not present with jquery 2.x but with 3.x. maybe related to https://jquery.com/upgrade-guide/3.0/#breaking-change-document-ready-handlers-are-now-asynchronous

**Comment by webzwo0i** [comment]
when I understand it right https://github.com/ether/etherpad-lite/blob/develop/src/static/js/collab_client.js#L761 is run before the onload-handler is triggered that runs padeditbar.init https://github.com/ether/etherpad-lite/blob/develop/src/static/js/ace.js#L307

**Comment by webzwo0i** [comment]
``diff --git a/src/static/js/collab_client.js b/src/static/js/collab_client.js
index 0031354f..66fce90e 100644
--- a/src/static/js/collab_client.js
+++ b/src/static/js/collab_client.js
@@ -758,7 +758,7 @@ function getCollabClient(ace2editor, serverVars, initialUserInfo, options, _pad)
     setIsPendingRevision: setIsPendingRevision
   };
 
-  $(document).ready(setUpSocket);
+  setTimeout(setUpSocket(),0);
   return self;
 }
``
fixes it for me. frontend tests pass, but probably there are no tests for reconnect/connect/disconnect modals?

**Comment by webzwo0i** [comment]
also present on https://video.etherpad.com/tests/frontend for me

**Comment by JohnMcLear** [comment]
@webzwo0i were you sending through a pr to fix this?

**Comment by JohnMcLear** [comment]
Bump for pr for fix (I can't replicate still)

**Comment by webzwo0i** [comment]
reopening this and assigning it to myself
The reason is although a patch is in develop that seems to fix the problem for me and @rittmeier , the last days I discovered that reverting the change does not expose the bug again for me... That's really strange.
Also there is no clear explanation why the jquery update that made the ready-event async triggered this specific bug.

I get "Pad never loaded" errors during frontend tests on my machine lately and that's why I studied the page loading-code of etherpad a lot in the last days. I am not sure anymore that the fix in #4158 is correct - it might just be pure coincidence - so I leave this open for a moment to not forget about it
